app/services/proofing/socure/id_plus/request.rb

```diff
@@ -0,0 +1,158 @@
+# frozen_string_literal: true
+
+module Proofing
+  module Socure
+    module IdPlus
+      class RequestError < StandardError
+        def initialize(wrapped)
+          @wrapped = wrapped
+          super(build_message)
+        end
+
+        def reference_id
+          return @reference_id if defined?(@reference_id)
+          @reference_id = response_body.is_a?(Hash) ?
+            response_body['referenceId'] :
+            nil
+        end
+
+        def response_body
+          return @response_body if defined?(@response_body)
+          @response_body = if wrapped.respond_to?(:response_body)
+                             wrapped.response_body
+                           end
+        end
+
+        def response_status
+          return @response_status if defined?(@response_status)
+          @response_status = if wrapped.respond_to?(:response_status)
+                               wrapped.response_status
+                             end
```

Comment:
same deal with `.try` or maybe even the trailing format?

```suggestion
          @response_status = wrapped.try(:response_status)
```

or

```suggestion
          @response_status = wrapped.response_status if wrapped.respond_to?(:response_status)
```

Reply:
TIL about `.try`